### PR TITLE
More documentation improvements

### LINKED ISSUE #43: Further docs improvements
Before diving into explicit code examples, a introduction to the theoretical reduced basis framework as well as the application to quantum spin systems would be nice.

On the one hand many readers might not be familiar with reduced basis methods but also people outside of physics probably find many body Hamiltonians confusing at first glance. The reduced basis part would consider general parametrized eigenvalue problems and explain the basics of the greedy algorithm. The construction of the spin Hamiltonians could already be shifted to this introduction, such that the beginning of the "The reduced basis workflow" example doesn't have to deal with introducing spin physics first.

### PR DESCRIPTION
**Pull request by pbrehmer**
Corrects minor issues and makes language consistent with [recent paper](https://arxiv.org/abs/2304.13587).

- closes #43

### CODE REVIEW

File: src/rbasis.jl [hunk 1]
```diff
@@ -54,31 +54,36 @@ end
...
-        overlaps = map(CartesianIndices((1:length(v1), 1:length(v2)))) do idx
-            dot(v1[first(idx.I)], v2[last(idx.I)])
+        overlaps = map(Iterators.product(1:length(v1), 1:length(v2))) do (i, j)
+            dot(v1[i], f(v2[j]))
         end
         return overlaps
     end
 end
 
+overlap_matrix(v1::Vector, v2::Vector) = overlap_matrix(x -> x, v1, v2)
```
Comment (mfherbst) [inline src/rbasis.jl:85]: Use identity instead of x -> x ?

File: src/affine_decomposition.jl [hunk 2]
```diff
@@ -99,7 +99,14 @@ end
 Compress one term of an [`AffineDecomposition`](@ref) `ApproxMPO` type.
 """
 function compress(op, snapshots::AbstractVector)
-    overlap_matrix(snapshots, map(Ψ -> op * Ψ, snapshots))
+    overlap_matrix(Ψ -> op * Ψ, snapshots, snapshots)
+    # M = length(snapshots)
+    # matrixel = Matrix{Number}(undef, M, M)
```
Comment (mfherbst) [inline src/affine_decomposition.jl:104]: Remove comments?

**Review by mfherbst — COMMENTED** [review]
Lgtm

File: src/rbasis.jl (hunk 1 above)
Comment (pbrehmer) [reply src/rbasis.jl]: Didn't know that existed, surely is more fitting here

**Comment by pbrehmer** [comment]
I'll still make some minor modifications to the text and then we should be ready to merge.

**Review by mfherbst — APPROVED** [review]

**Pull request merged**